logstreams/src/main/java/io/zeebe/logstreams/spi/SnapshotController.java

```diff
@@ -51,7 +52,7 @@
   void replicateLatestSnapshot(Consumer<Runnable> executor);
 
   /** Registers to consumes replicated snapshots. */
-  void consumeReplicatedSnapshots(Consumer<Long> dataDeleteCallback);
+  void consumeReplicatedSnapshots(DeletionService deletionService);
```

Comment:
This looks weird to me.

Reply:
But I have no different solution on the fly. I'm not sure whether my problem is the naming or the coupling :sweat_smile: Maybe it would make it a bit better if we rename it to something like `triggerDeletionOnReceivingMaxSnapshotCount` or you split it into two methods to set the deletion service and the other one starts the consuming.

Maybe we can remove the coupling of the replication and the deletion, such that the SnapshotController manages the deletion and is triggered when a valid snapshot was received.